src/ansys/hps/client/client.py

```diff
@@ -225,13 +236,39 @@ def __init__(
         self._unauthorized_num_retry = 0
         self._unauthorized_max_retry = 1
 
+        def exit_handler():
+            log.info("Exiting gracefully.")
+            if self.dt_client is not None:
+                self.dt_client.stop()
+
+        atexit.register(exit_handler)
+
     @property
     def rep_url(self) -> str:
         msg = "The client 'rep_url' property is deprecated. Use 'url' instead."
         warnings.warn(msg, DeprecationWarning)
         log.warning(msg)
         return self.url
 
+    def _start_dt_worker(self):
+
+        if self.dt_client is None:
+            log.info("Starting Data Transfer client.")
+            # start Data transfer client
+            self.dt_client = DTClient()
```

Comment:
You should set the `download_dir` to what we agreed for all clients:
- windows: ${LOCALAPPDATA}/Ansys/binaries/worker
- linux: ${HOME}/.ansys/binaries/worker
- macos: ask Michal what's preferable :-)

Reply:
updated code to use proper directory for downloading binary